### DISCUSSION TITLE
Python version & application in ERP analyses

### DISCUSSION THREAD

**Discussion by teanijarv** [lucwilson/SPRiNT#3] [Q&A] [answered]
Just wanted to ask, what is the current status on the Python version of SPRiNT? Is it ready to use or are you still working on it and is it close to being ready or you would recommend me to use the MATLAB code instead?

Secondly, more analysis-related question not directly related to your project - would analysis with SPRiNT be eligible to use in ERP studies too? For example, I have 1-second epochs (-0.2s-0.8s), so I should change the STFT window to be around 200-250ms - would it be long enough window to perform spectral parametrisation with your method?

Thanks for this great project!

**Comment by lucwilson** [comment, marked as answer]
Hi there,

Thank you for your interest in SPRiNT! A few weeks ago, I [added a new script](https://github.com/lucwilson/SPRiNT/commit/eaa86d02fa00e793cbca27cb5fa9b3b052e9e031) to our Github titled "SPRiNT_py.py" containing all the necessary code to run SPRiNT and peak post-processing in Python. Both versions of SPRiNT should give the same results (+/- a few thousandths of difference in parameters), though you may find the MATLAB version to be slightly more refined (I am admittedly less experienced with Python).

As for your question of analysis, a 1 s epoch would be really pressing on the limits for the STFT. For instance, using a 200-250 ms sliding window would result in 4-5 Hz frequency resolution, which is likely insufficient to identify peaks in the spectrogram (though it may still be able to parameterize the aperiodic component). Furthermore, exploring aperiodic activity in ERPs is a slightly difficult issue. For instance, averaging trial time series should only preserve the time-locked elements of brain activity, which arguably shouldn't be aperiodic. [Gyurkovics et al. (2021)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9480870/) explore this matter in more detail in the context of an attention task (highly recommended read). Part of the solution could be to average trials at the level of the spectrogram, but here it is still important to distinguish between changes in aperiodic activity induced vs. evoked by your event.

I hope this helps with your analyses! If you have any other questions, please feel free to reach out. :)

Luc

**Reply by teanijarv** [reply]
Thank you very much! You made it much more clearer for me now.

Also, thanks for sharing this paper with me. It gave me more interesting ideas to try for my analysis.

Best,
Toomas
